src/ngraph/op/fused/group_conv_transpose.cpp

```diff
@@ -0,0 +1,320 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <iterator>
+#include <numeric>
+
+#include "ngraph/op/concat.hpp"
+#include "ngraph/op/convolution.hpp"
+#include "ngraph/op/fused/group_conv_transpose.hpp"
+#include "ngraph/op/slice.hpp"
+#include "ngraph/util.hpp"
+#include "ngraph/validation_util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::GroupConvolutionTranspose::GroupConvolutionTranspose(const shared_ptr<Node>& data,
+                                                         const shared_ptr<Node>& filters,
+                                                         const Strides& strides,
+                                                         const Strides& dilations,
+                                                         const CoordinateDiff& padding_begin,
+                                                         const CoordinateDiff& padding_end,
+                                                         const CoordinateDiff& output_padding,
+                                                         const size_t groups,
+                                                         const PadType& pad_type,
+                                                         const Shape& output_shape)
```

Comment:
Could you explain why `output_shape` needs to be a parameter here? Normally we would infer this.

Reply:
That is actually ONNX specific. Since it enables to define `output_shape` and let infer required padding for this.

Reply:
It confused me as well but i assume ONNX had a reason for going down this path.